crates/worker/src/partition/cleaner.rs

```diff
@@ -281,6 +281,9 @@ mod tests {
         )
         .unwrap();
 
+        // By yielding once we let the cleaner task run, and perform the cleanup
+        tokio::task::yield_now().await;
```

Comment:
What I don't like about this test is that it depends on the fact that `Interval` immediately triggers and that the task has been registered when we reach this point. Maybe it is what it is but if we had the clock used by the cleaner under our control, then we could make the test more robust.

Reply:
I think this is ok for the time being